// frog@0.18.3/_lib/utils/getComposerActionContext.js
export function getComposerActionContext(parameters) {
    const { context } = parameters;
    const { env, actionData, req, verified } = context || {};
    if (!actionData)
        throw new Error('Action data must be present for action handlers.');
    if (!actionData.state)
        throw new Error('State must be present for composer action handler.');
    return {
        context: {
            actionData: {
                buttonIndex: 1,
                fid: actionData.fid,
                network: actionData.network,
                messageHash: actionData.messageHash,
                timestamp: actionData.timestamp,
                state: actionData.state,
                url: actionData.url,
            },
            env,
            error: (data) => ({
                error: data,
                format: 'composerAction',
                status: 'error',
            }),
            req,
            res: (data) => ({
                data,
                format: 'composerAction',
                status: 'success',
            }),
            var: context.var,
            verified,
        },
    };
}
