// frog@0.18.3/_lib/utils/verifyComposerAction.js
import { bytesToHex, bytesToString } from 'viem';
import { verifyMessage } from './verifyMessage.js';
export async function verifyComposerAction(parameters) {
    const { message } = await verifyMessage(parameters);
    const composerActionData = messageToComposerActionData(message);
    return { composerActionData };
}
////////////////////////////////////////////////////////////////////
// Utilities
export function parseComposerActionDataState(state) {
    return JSON.parse(decodeURIComponent(state));
}
export function messageToComposerActionData(message) {
    const frameActionBody = message.data?.body.value;
    const composerActionData = {
        fid: Number(message.data?.fid),
        messageHash: bytesToHex(message.hash),
        network: message.data?.network,
        timestamp: message.data?.timestamp,
        state: parseComposerActionDataState(bytesToString(frameActionBody.state)),
        url: bytesToString(frameActionBody.url),
        buttonIndex: frameActionBody.buttonIndex,
    };
    return composerActionData;
}
