// frog@0.18.3/_lib/utils/verifyFrame.js
import { bytesToHex, bytesToString } from 'viem';
import { verifyMessage } from './verifyMessage.js';
export async function verifyFrame(parameters) {
    const { message } = await verifyMessage(parameters);
    const frameData = messageToFrameData(message);
    return { frameData };
}
////////////////////////////////////////////////////////////////////
// Utilties
export function messageToFrameData(message) {
    const frameActionBody = message.data?.body.value;
    const frameData = {
        address: frameActionBody.address
            ? bytesToHex(frameActionBody.address)
            : undefined,
        castId: {
            fid: Number(frameActionBody.castId?.fid),
            hash: bytesToHex(frameActionBody.castId?.hash),
        },
        fid: Number(message.data?.fid),
        messageHash: bytesToHex(message.hash),
        network: message.data?.network,
        timestamp: message.data?.timestamp,
        url: bytesToString(frameActionBody.url),
        buttonIndex: frameActionBody.buttonIndex,
        inputText: bytesToString(frameActionBody.inputText),
        state: bytesToString(frameActionBody.state),
        transactionId: frameActionBody.transactionId
            ? bytesToHex(frameActionBody.transactionId)
            : undefined,
    };
    return frameData;
}
